Feature/dataset: Dataset creation

evaplaysviolin opened the pull request:
Created a base set of pairs (10 words and their translations in French)
Wrote preliminary function for generating dataset from raw data
Output for `datasetMap` should be the Map we will work with for game play?

Blockers:

- Looking for review on whether this is anywhere near correct approach

evaplaysviolin committed:
Create dataset using Map for word pairs

serpient commented on src/VocabGameData.js:
```diff
@@ -0,0 +1,39 @@
+function createDataset(base) {
+  base.forEach(function(array) {
+    dataset.push([
+      array[0],
```
are you trying to use pure arrays, or a MAP object here?

serpient replied:
oh i see, you are making it an array first and then converting to a MAP object at the end.

evaplaysviolin replied:
Yes, but this isn't any particular decision I made, it's because I didn't know if I could do it directly with MAP

serpient reviewed (commented):
your code looks good -- 
im going into this branch and push some tests to make sure everything is as expected,
then we can probably approve.

serpient commented on src/VocabGameData.js:
```diff
@@ -0,0 +1,39 @@
+}
+
+createDataset(basePairs);
+let datasetMap = new Map(dataset);
```
ok, I would tuck in the new Map() conversion within the function `createDataset`

so then all we have  to do is run the function, and it spits out the Map dataset without doing another step.

serpient replied:
let me know when u made the change and ill pull!

evaplaysviolin replied:
@serpient I made the changes!

evaplaysviolin committed:
Update mapping function to return MAP directly

serpient committed:
add test for dataset type

serpient committed:
Merge branch 'feature/dataset' of https://github.com/chingu-voyages/Vocab-Study-Buddy into feature/dataset

serpient commented on src/VocabGameData.js:
```diff
@@ -0,0 +1,40 @@
+  ["can you help me, please?", "pouvez-vous m’aider, s’il vous plaît?"]
+];
+
+let datasetMap;
```
this doesnt need to be defined in global scope as its only used in the function

serpient commented on src/VocabGameData.js:
```diff
@@ -0,0 +1,40 @@
+      }
+    ]);
+  });
+  datasetMap = new Map(dataset);
```
you can just `return new Map(dataset)` here and not keep track of datasetMap at all.

essentially, you are using datasetMap just to return the value, so technically, its not necessary.

also i think you forgot to return the dataset on this line.

serpient reviewed (changes requested):
just a few little things!

serpient commented on src/VocabGameData.js:
```diff
@@ -0,0 +1,40 @@
+  datasetMap = new Map(dataset);
+}
+
+createDataset(basePairs);
```
i mean here, we would define it like so
` export const datasetMap = createDataset(basePairs) `

so we initialize this variable here and export it at the same time.

evaplaysviolin replied:
Okay, I think I understand, let me make these changes

evaplaysviolin committed:
Actually update function to return MAP, add exports

serpient committed:
Merge branch 'feature/dataset' of https://github.com/chingu-voyages/Vocab-Study-Buddy into feature/dataset

serpient commented:
to run tests, you can run 
`npm test --watchAll` in a different terminal window

serpient committed:
check that word pairs are stored correctly in mapped object

serpient committed:
remove unecesasry testing of official Map functions (like set/get)

this is testing the official methods of the MAP data type defined from javascript. its unnecessary.

serpient committed:
add error handling unit tests

serpient commented:
@evaplaysviolin 

these are the Error handling cases your function should handle (not sure if you get the notifications like I do in VSCODE, possibly this is the Jest extension)
![image](https://user-images.githubusercontent.com/29721784/49772371-e82d0980-fca1-11e8-9a2d-3972ccd505d2.png)

evaplaysviolin committed:
Correct initialized values for isSelected/isCorrect from null to false

evaplaysviolin committed:
Merge branch 'feature/dataset' of https://github.com/chingu-voyages/Vocab-Study-Buddy into feature/dataset
